crates/sui-benchmark/tests/simtest.rs

```diff
@@ -249,10 +257,118 @@ mod test {
         assert_eq!(expected_checkpoint, pruned);
     }
 
+    #[sim_test(config = "test_config()")]
+    async fn test_testnet_upgrade_compatibility() {
+        // This test is intended to test the compatibility of the latest protocol version with
+        // all previous protocol versions on testnet. It does this by starting a testnet with
+        // the old protocol version that this binary supports, and then upgrading it all the
+        // way to the latest protocol version.
+        let min_ver = ProtocolVersion::MIN.as_u64();
+        let max_ver = ProtocolVersion::MAX.as_u64();
+        let init_framework =
+            sui_framework_snapshot::load_bytecode_snapshot("testnet", min_ver).unwrap();
+        let mut test_cluster = init_test_cluster_builder(7, 5000)
+            .with_protocol_version(ProtocolVersion::MIN)
+            .with_supported_protocol_versions(SupportedProtocolVersions::new_for_testing(
+                min_ver, min_ver,
+            ))
+            .with_fullnode_supported_protocol_versions_config(
+                SupportedProtocolVersions::new_for_testing(min_ver, max_ver),
+            )
+            .with_objects(init_framework)
+            .build()
+            .await
+            .unwrap();
+
+        // let test_init_data = TestInitData::new(&test_cluster).await;
+
+        let finished = Arc::new(AtomicBool::new(false));
+        let finished_clone = finished.clone();
+        let _handle = tokio::task::spawn(async move {
+            for version in min_ver..=max_ver {
+                info!("Targeting protocol version: {}", version);
+                for h in test_cluster.all_node_handles() {
```

Comment:
We have `test_cluster.wait_for_epoch`. It might be worth implementing a `wait_for_protocol_version` using similar idea and use it here.